specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,400 @@
+* The developer will first obtain a Remote Identifer GUID and a client secret as part of this Portal Registration flow. This is a one time operation by the developer to provision an app. 
+* When requesting a channel using WNS client Reunion APIs, the developer will pass in a remoteId and WNS will return a specially encoded Channel URL with that Remote Identifier. 
+* When the application service Pushes a notification down to the client, it will retrieve an access token with AAD using the client secret provisioned above. 
+* The application service will then pass that access token along with the payload in a Http POST operation to the channel URL. 
+* WNS will ensure that the payload is ultimately delivered to the app residing on the client device.
+
+Phase 1 (Reunion Preview):
+WNS Push APIs will only support WIN32 Packaged app (MSIX).
+
+Phase 2 (Reunion V1):
+Our Push APIs will add support for Unpackaged Win32 scenarios.
+
+# Examples
+
+## In this scenario, the process that Registers the Push Activator and the process specified as the COM server are the same
+The code in Main would follow the pattern below:
+* The Registration will take in a CLSID of the COM server as part of the Activator setup. The registration API will simply be a thin wrapper around the BackgroundInfra WinRT APIs and abstract away the COM details from the developer including registration of the inproc Reunion component.
+* The app will query ApplifeCycle APIs to retrieve an ActivationKind. (Note: This is covered in a seperate AppLifeCycle API spec.)
+* If the Activation is Push, the app will QI the Arguments property to retrieve an instance of PushActivatedEventArgs and get the Push payload from it. Care needs to be taken to Take a Deferral and Release the Deferral while processing the payload for Power Management compliance.
+* If the Activation is Foreground, the app will do 2 things:
+  * It will subscribe to a Foreground In-memory Push event
+  * It will request a Push Channel Asynchronously with an implementation of Progress and Completed event handler.
+    * Expect Channel operations to take around 2-16 minutes in some rare cases (retryable errors). In the 90th percentile case, it should be fairly quick operation (in a few seconds).  
+
+```cpp
+int main()
+{
+    PushActivatorDetails details(winrt::guid("BACCFA91-F1DE-4CA2-B80E-90BE66934EC6"));
+
+    // Registers a Push Trigger and sets up an inproc COM Server for Activations
+    PushManager::RegisterActivator(details, PushRegistrationType::PushTrigger | PushRegistrationType::ComActivator);
+
+    // Check to see if the WinMain activation is due to a Push Activator
+    // Note: This API is currently not present in Reunion but will be included as part of the AppLifecycle Public API spec.
+    auto activation = AppLifecycle::Activation();
+    auto kind = activation.Kind();
+
+    if (kind == ReunionActivationKind::Push)
+    {
+        auto pushArgs = activation.Arguments().try_as<PushReceivedEventArgs>();
```

Comment:
Should be `as<>()` so your telemetry will capture the QI failure reason, rather than just crashing on a null pointer at the next line and not knowing why the QI failed.